src/GraphQL.MicrosoftDI/ScopedConnectionBuilderExtensions.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Threading.Tasks;
+using GraphQL.Builders;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.MicrosoftDI
+{
+    /// <summary>
+    /// Extension methods for creating connection resolver builders.
+    /// </summary>
+    public static class ScopedConnectionBuilderExtensions
+    {
+        /// <summary>
+        /// Sets the resolver for the connection field. A dependency injection scope is created for the duration of the resolver's execution
+        /// and the scoped service provider is passed within <see cref="IResolveFieldContext.RequestServices"/>.
+        /// </summary>
+        public static ConnectionBuilder<TSourceType> ResolveScoped<TSourceType, TReturnType>(this ConnectionBuilder<TSourceType> builder, Func<IResolveConnectionContext<TSourceType>, TReturnType> resolver)
+        {
+            if (resolver == null)
+                throw new ArgumentNullException(nameof(resolver));
+            builder.Resolve(context =>
+            {
+                using (var scope = (context.RequestServices ?? throw new MissingRequestServicesException()).CreateScope())
+                {
+                    return resolver(new ScopedResolveConnectionContextAdapter<TSourceType>(context, scope.ServiceProvider));
+                }
+            });
+            return builder;
+        }
+
+        /// <inheritdoc cref="ResolveScoped{TSourceType, TReturnType}(ConnectionBuilder{TSourceType}, Func{IResolveConnectionContext{TSourceType}, TReturnType})"/>
+        public static ConnectionBuilder<TSourceType> ResolveScopedAsync<TSourceType, TReturnType>(this ConnectionBuilder<TSourceType> builder, Func<IResolveConnectionContext<TSourceType>, Task<TReturnType>> resolver)
+        {
+            if (resolver == null)
+                throw new ArgumentNullException(nameof(resolver));
+            builder.ResolveAsync(async context =>
+            {
+                using (var scope = (context.RequestServices ?? throw new MissingRequestServicesException()).CreateScope())
+                {
+                    return await resolver(new ScopedResolveConnectionContextAdapter<TSourceType>(context, scope.ServiceProvider));
+                }
+            });
+            return builder;
+        }
+
+        /// <summary>
+        /// Creates a resolve builder for the connection field.
+        /// </summary>
+        public static ConnectionResolverBuilder<TSourceType, object> Resolve<TSourceType>(this ConnectionBuilder<TSourceType> builder)
```

Comment:
Will this method then be deleted along with the `ConnectionBuilder<TSourceType>`?

Reply:
I expect so.